<!-- -*- mode: XML -*- -->
<!-- run velodyne_pointcloud/CloudNodelet in a nodelet manager for a VLP-16 -->

<launch>

  <!-- declare arguments with default values -->
  <arg name="calibration" default="$(find velodyne_pointcloud)/params/VLP16db.yaml"/>
  <arg name="device_ip" default="" />
  <arg name="frame_id" default="velodyne" />
  <arg name="manager" default="$(arg frame_id)_nodelet_manager" />
  <arg name="max_range" default="130.0" />
  <arg name="min_range" default="0.4" />
  <!--put the pcap and imu file in the same directory-->
  <arg name="pacp_imu_dir" default="/home/llw/data1/Downloads/datavlp16/20191231_154515"/>
  <arg name="pcap" default="$(arg pacp_imu_dir)/0_lidar.pcap" />
  <arg name="packets_per_scan" default="1" />
  <arg name="port" default="" />
  <arg name="read_fast" default="true" />
  <arg name="read_once" default="false" />
  <arg name="use_pcap_utc_time" default="true" />
  <arg name="repeat_delay" default="0.0" />
  <arg name="rpm" default="600.0" />
  <arg name="gps_time" default="true" />
  <arg name="cut_angle" default="-0.01" />
  <arg name="timestamp_first_packet" default="false" />
  <arg name="laserscan_ring" default="-1" />
  <arg name="laserscan_resolution" default="0.007" />
  <arg name="organize_cloud" default="false" />

  <!-- start nodelet manager and driver nodelets -->
  <include file="$(find velodyne_driver)/launch/nodelet_manager.launch">
    <arg name="device_ip" value="$(arg device_ip)"/>
    <arg name="frame_id" value="$(arg frame_id)"/>
    <arg name="manager" value="$(arg manager)" />
    <arg name="model" value="VLP16"/>
    <arg name="pcap" value="$(arg pcap)"/>
    <arg name="port" value="$(arg port)"/>
    <arg name="read_fast" value="$(arg read_fast)"/>
    <arg name="read_once" value="$(arg read_once)"/>
    <arg name="use_pcap_utc_time" value="$(arg use_pcap_utc_time)" />
    <arg name="repeat_delay" value="$(arg repeat_delay)"/>
    <arg name="rpm" value="$(arg rpm)"/>
    <arg name="gps_time" value="$(arg gps_time)"/>
    <arg name="cut_angle" value="$(arg cut_angle)"/>
    <arg name="timestamp_first_packet" value="$(arg timestamp_first_packet)"/>
    <arg name="packets_per_scan" value="$(arg packets_per_scan)" />
  </include>

  <!-- start cloud nodelet -->
  <include file="$(find velodyne_pointcloud)/launch/cloud_nodelet.launch">
    <arg name="model" value="VLP16"/>
    <arg name="calibration" value="$(arg calibration)"/>
    <arg name="manager" value="$(arg manager)" />
    <arg name="max_range" value="$(arg max_range)"/>
    <arg name="min_range" value="$(arg min_range)"/>
    <arg name="organize_cloud" value="$(arg organize_cloud)"/>
  </include>

  <!-- start recordpoints_node -->
  <node pkg="velodyne_pointcloud" type="recordpoints_node" name="recordpoints_node" output="screen">
    <param name="bag_dir" value="/home/llw/data1/bag/" />
    <param name="topic_name_lidar" value="velodyne_points" />
    <param name="topic_name_imu" value="imu" />
  </node>

  <!-- start convert_imu_node -->
  <node pkg="velodyne_pointcloud" type="convert_imu_node" name="convert_imu_node" output="screen">
    <param name="imu_file" value="$(arg pacp_imu_dir)/remote.imu" />
    <param name="imu_frame_ID" value="imu" />
    <param name="imu_type" value="ADIS16465" />
    <param name="imu_freq_Hz" value="250" />
    <param name="imr_header" value="false" />
  </node>
</launch>
